packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -7022,4 +7024,105 @@ describe("Teal Opcodes", function () {
 			expectRuntimeError(() => new Base64Decode([], 0), RUNTIME_ERRORS.TEAL.ASSERT_LENGTH);
 		});
 	});
+
+	describe("Tealv7: replace2 opcode", function () {
+		let stack: Stack<StackElem>;
+		this.beforeEach(() => {
+			stack = new Stack<StackElem>();
+		});
+
+		it("Should replace bytes correctly", () => {
+			const original = "0x11111111";
+			const replace = "0x2222";
+			let hexStr = "0x22221111";
+			let expectedBytes = new Uint8Array(Buffer.from(hexStr.slice(2), "hex"));
+
+			stack.push(new Uint8Array(Buffer.from(original.slice(2), "hex")));
+			stack.push(new Uint8Array(Buffer.from(replace.slice(2), "hex")));
+			let op = new replace2(["0"], 0);
+			op.execute(stack);
+			assert.deepEqual(stack.pop(), expectedBytes);
```

Comment:
you should check the whole stack, not only the top element.

Reply:
shouldn't we check if the opcode replace2 and replace3 run corrrectly right?
so why we should check the whole stack?

Reply:
because this will  be the proper test - you want to make sure that only 2 top elements changed, and rest is the same. Now you only test that the top element changed.

Reply:
oh I see.
let me add a bytes to the stack and check if that bytes remain unchanged!